source/fuzz/transformation_split_block.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_split_block.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "source/util/make_unique.h"
+
+namespace spvtools {
+namespace fuzz {
+
+using opt::BasicBlock;
+using opt::IRContext;
+using opt::Instruction;
+using opt::Operand;
+
+std::pair<bool, BasicBlock::iterator>
+TransformationSplitBlock::FindInstToSplitBefore(BasicBlock* block) {
+  // There are three possibilities:
+  // (1) the transformation wants to split at some offset from the block's
+  //     label.
+  // (2) the transformation wants to split at some offset from a
+  //     non-label instruction inside the block.
+  // (3) the split associated with this transformation has nothing to do with
+  //     this block
+  if (result_id_ == block->id()) {
+    // Case (1).
+    if (offset_ == 0) {
+      // The offset is not allowed to be 0: this would mean splitting before the
+      // block's label.
+      return {true, block->end()};
```

Comment:
This is a confusing result for the comment. Shouldn't it return {false, block-end()}?

Reply:
I have clarified the reason as follows:

      // The offset is not allowed to be 0: this would mean splitting before the
      // block's label.
      // By returning (true, block->end()), we indicate that we did find the
      // instruction (so that it is not worth searching further for it), but
      // that splitting will not be possible.

and also there is a comment at the top of the method:

// Returns:
// - (true, block->end()) if the relevant instruction is in this block
//      but inapplicable
// - (true, it) if 'it' is an iterator for the relevant instruction
// - (false, _) otherwise.